shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -60,9 +60,19 @@
     return;
   }
 
-  // TODO(amirh): decode and pass the creation args.
+  id params = nil;
+  if ([factory respondsToSelector:@selector(createArgsCodec)]) {
+    NSObject<FlutterMessageCodec>* codec = [factory createArgsCodec];
+    if (codec != nil && args[@"params"] != nil) {
+      FlutterStandardTypedData* paramsData = args[@"params"];
+      params = [codec decode:paramsData.data];
+    }
+  }
+
   FlutterTouchInterceptingView* view = [[[FlutterTouchInterceptingView alloc]
-      initWithEmbeddedView:[factory createWithFrame:CGRectZero viewIdentifier:viewId arguments:nil]
+      initWithEmbeddedView:[factory createWithFrame:CGRectZero
```

Comment:
Nit: This line is getting a bit long, can you add that embedded view as a variable so its easier to read?

Reply:
done